src/test/java/com/thealgorithms/strings/AhoCorasickTest.java

```diff
@@ -0,0 +1,138 @@
+/****************************************************
+ * Tests For Aho-Corasick String Matching Algorithm *
+ *                                                  *
+ *            Author: Prabhat-Kumar-42              *
+ *    GitHub: https://github.com/Prabhat-Kumar-42   *
+ ****************************************************/
+
+package com.thealgorithms.strings;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.ArrayList;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class AhoCorasickTest {
+    private String[] patterns;
+    private String text;
+
+    @BeforeEach
+    void setUp() {
+        patterns = new String[] {"ACC", "ATC", "CAT", "GCG", "C", "T"};
+        text = "GCATCG";
+    }
+
+    @Test
+    void testSearch() {
+        // Test searching for multiple patterns in the input text.
+        Map<String, ArrayList<Integer>> positions = AhoCorasick.search(text, patterns);
+
+        // Define expected results for each pattern
+        int[][] expected = {
+            {}, // "ACC" should not be found
+            {2}, // "ATC" should be found at index 2
+            {1}, // "CAT" should be found at index 1
+            {}, // "GCG" should not be found
+            {1, 4}, // "C" should be found at indices 1 and 4
+            {3} // "T" should be found at index 3
+        };
+
+        // Check specific pattern occurrences
+        for (int i = 0; i < patterns.length; i++) {
+            assertIterableEquals(toList(expected[i]), positions.get(patterns[i]));
+        }
+    }
```

Comment:
Please notice that this test could still pass, if the `search` method would return a map with some additional keys.

What do you think about the following:
```suggestion
    @Test
    void testSearch() {
        final var expected = Map.of("ACC", new ArrayList<>(Arrays.asList()), "ATC", new ArrayList<>(Arrays.asList(2)), "CAT", new ArrayList<>(Arrays.asList(1)), "GCG", new ArrayList<>(Arrays.asList()), "C", new ArrayList<>(Arrays.asList(1, 4)), "T", new ArrayList<>(Arrays.asList(3)));

        assertEquals(expected, AhoCorasick.search(text, patterns));
    }
```

In case of normal line lengths, the code above looks really nice (in my opinion).

Make sure to
```java
import java.util.Arrays;
```

Please apply above idea in other tests.

Reply:
yes, got it, will implement it.